ci/cpu/build.sh

```diff
@@ -80,6 +80,15 @@ fi
 if [ "$BUILD_LIBCUDF" == '1' ]; then
   gpuci_logger "Build conda pkg for libcudf"
   gpuci_conda_retry mambabuild --no-build-id --croot ${CONDA_BLD_DIR} conda/recipes/libcudf $CONDA_BUILD_ARGS
+
+  # BUILD_LIBCUDF == 1 means this job is being run on the cpu_build jobs
+  # that is where we must also build the strings_udf package
+  gpuci_logger "Build conda pkg for strings_udf (python 3.8)"
+  gpuci_conda_retry mambabuild --no-build-id --croot ${CONDA_BLD_DIR} conda/recipes/strings_udf $CONDA_BUILD_ARGS --python=3.8
```

Comment:
You should use `--python=$PYTHON` and only perform a single build. CI will handle running the multiple python versions.

Reply:
Hi @raydouglass , I gave this a shot in https://github.com/rapidsai/cudf/pull/11319/commits/1f8cb3baacc823a5fce6fa2f9342a75bc361e9fd but it seemed to result in a failure to build for both python 3.8 and 3.9 in the GPU test jobs. Example here:

https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-gpu-test/CUDA=11.5,GPU_LABEL=driver-495,LINUX_VER=ubuntu20.04,PYTHON=3.9/10519/console

```
package strings_udf-22.10.00a-cuda_11_py38_g195998cead_471 requires python >=3.8,<3.9.0a0, but none of the providers can be installed
```

Reply:
@raydouglass I remembered why this was set up this way in the first place. The `strings_udf` package compiles `libcudf` code as part of its conda build process. This means that because `libcudf` requires cuda 11.5 to compile, `strings_udf` also inherits that requirement. As such, we choose to build the conda packages only under situations where `BUILD_LIBCUDF == 1`, letting CI do the job of making sure that only occurs with cuda 11.5. 

IIUC, our CI doesn't bother building `libcudf` with varying `$PYTHON`, because it contains no python by itself. Thus if we use `$PYTHON` here, we end up missing a package. The simplest solution seemed to be to build both explicitly.

Reply:
I'll mark this as resolved. I think Brandon did a pretty good job of summing up why we decided to build both Python versions in the C++ build stage. It's a hack for sure, but it's a workaround for the limitations in our current CI setup and will go away in the near future when we switch to GitHub Actions.